pkg/handler/processor/ite6/ite6_test.go

```diff
@@ -0,0 +1,254 @@
+				},
+			},
+			{
+				URI: "github_hosted_vm:ubuntu-18.04:20210123.1",
+			},
+		},
+	}
+)
+
+func TestITE6Processor_ValidateSchema(t *testing.T) {
+	tests := []struct {
+		name    string
+		args    *processor.Document
+		wantErr bool
+	}{
+		{
+			name: "ITE6 Doc with unknown payload",
+			args: &processor.Document{
+				Blob:   []byte(badProvenance),
+				Type:   processor.DocumentITE6,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: "TestCollector",
+					Source:    "TestSource",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "ITE6 Doc with valid payload",
+			args: &processor.Document{
+				Blob:   []byte(ite6SLSA),
+				Type:   processor.DocumentITE6,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: "TestCollector",
+					Source:    "TestSource",
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &ITE6Processor{}
+			if err := e.ValidateSchema(tt.args); (err != nil) != tt.wantErr {
+				t.Errorf("ITE6Processor.ValidateSchema() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestITE6Processor_Unpack(t *testing.T) {
+	predicatePayload, _ := json.Marshal(predicate)
+	tests := []struct {
+		name    string
+		args    *processor.Document
+		want    []*processor.Document
+		wantErr bool
+	}{
+		{
+			name: "ITE6 Doc with unknown payload",
+			args: &processor.Document{
+				Blob:   []byte(unknownProvenance),
+				Type:   processor.DocumentITE6,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: "TestCollector",
+					Source:    "TestSource",
+				},
+			},
+			want: []*processor.Document{{
+				Blob:   predicatePayload,
+				Type:   processor.DocumentUnknown,
+				Format: processor.FormatUnknown,
+				SourceInformation: processor.SourceInformation{
+					Collector: "TestCollector",
+					Source:    "TestSource",
+				},
+			}},
+			wantErr: false,
+		},
+		{
+			name: "ITE6 Doc with valid payload",
+			args: &processor.Document{
+				Blob:   []byte(ite6SLSA),
+				Type:   processor.DocumentITE6,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: "TestCollector",
+					Source:    "TestSource",
+				},
+			},
+			want: []*processor.Document{{
+				Blob:   predicatePayload,
+				Type:   processor.DocumentSLSA,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: "TestCollector",
+					Source:    "TestSource",
+				},
+			}},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &ITE6Processor{}
+			got, err := e.Unpack(tt.args)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ITE6Processor.Unpack() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			parsedGot := parseV02(got[0].Blob)
```

Comment:
add check for `len(got)`, since it is ok according to the interface to return a len 0 slice.

Reply:
added.